classes/item.rb

```diff
@@ -1,18 +1,26 @@
+require 'date'
 class Item
-  def intialize(publish_date, archive)
+  attr_accessor :id, :publish_date, :archive
+
+  attr_reader :genre, :author, :source, :label
+
+  def initialize(publish_date, archive)
     @id = rand(1..1000)
     @genre = nil
     @author = nil
     @source = nil
     @label = nil
-    @publish_date = publish_date
+    @publish_date = Date.parse(publish_date)
     @archive = archive
   end
```

Comment:
as the requirement said `properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created.` instead of defining them in the initialize method we should either create a custom method for them or assign them in this way for example `attr_accessor :label `